sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -367,13 +442,177 @@ namespace Azure { namespace Storage { namespace DataLake {
+     *        the directory will be deleted. If "false" and the directory is non-empty, an error
+     *        occurs.
+     */
+    Azure::Core::Nullable<bool> RecursiveOptional;
+
+    /**
+     * @brief Specify the access condition for the path.
+     */
+    PathAccessConditions AccessConditions;
+  };
+
+  /**
+   * @brief Optional parameters for PathClient::GetProperties
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/getproperties
+   */
+  struct PathGetPropertiesOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief Specify the access condition for the path.
+     */
+    PathAccessConditions AccessConditions;
+  };
+
+  /**
+   * @brief Optional parameters for PathClient::GetAccessControl
+   */
+  struct PathAccessControlOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief Specify the access condition for the path.
+     */
+    PathAccessConditions AccessConditions;
+  };
+
+  /**
+   * @brief Optional parameters for PathClient::Read
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/read
+   */
+  struct FileReadOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief Specify the offset of the starting range to be retrieved.
+     */
+    Azure::Core::Nullable<int64_t> Offset;
+
+    /**
+     * @brief Specify the length to be retreived if an offset has been specified.
+     */
+    Azure::Core::Nullable<int64_t> Length;
+
+    /**
+     * @brief When this header is set to "true" and specified together with the Range header,
+     *        the service returns the MD5 hash for the range, as long as the range is less than
+     *        or equal to 4MB in size. If this header is specified without the Range header,
+     *        the service returns status code 400 (Bad Request). If this header is set to true
+     *        when the range exceeds 4 MB in size, the service returns status code 400 (Bad
+     *        Request).
+     */
+    Azure::Core::Nullable<bool> RangeGetContentMD5;
+
+    /**
+     * @brief Specify the access condition for the path.
+     */
+    PathAccessConditions AccessConditions;
+  };
+
+  /**
+   * @brief Optional parameters for FileClient::Create
+   */
+  struct FileRenameOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief Valid only when namespace is enabled. This parameter determines the behavior of the
```

Comment:
I guess namespace is always enabled when talking about datalake. So I don't think the first sentence is necessary.